### ISSUE TITLE
[Feature request] - Expose request_timeout parameter in the LM_Studio node

### ISSUE THREAD

**Issue by alessandroperilli** [xXAdonesXx/NodeGPT#22]
I added NodeGPT suite to the [AP Workflow 6.0](https://perilli.com/ai/comfyui). It's working very well, but some users are reporting a timeout error generated by ComfyUI if LM Studio doesn't respond quickly enough.

They solve the problem by increasing the `timeout_parameter` in `/ComfyUI/venv/lib/python3.11/site-packages/autogen/oai/completion.py`. Which is great, but not exactly convenient. 

I wonder if it's possible to expose that parameter in the LM_Studio node itself or, at least, increase its default for new installations.

Thank you.

**Comment by xXAdonesXx** [comment]
I finally added the timout feature. Also I added llama.cpp sothat it runs without LM Studio, but slower and only for the text generation node (see workflows).

**Comment by alessandroperilli** [comment]
Thank you! But I don't see the timeout parameter exposed in the LM_Studio node. Did you opt for a different implementation?

**Comment by xXAdonesXx** [comment]
Forgot to commit changes :). Now its updated on github. I have also added llava support today.

**Comment by alessandroperilli** [comment]
I just updated and there are 2 problems:

1. I still don't see the timeout option exposed in the `LM_Studio` node UI (assuming that it's how you implemented it):

<img width="874" alt="Screenshot 2023-12-01 at 09 01 39" src="https://github.com/xXAdonesXx/NodeGPT/assets/9775949/de998a39-cd80-4876-837c-c84293d200e7">

2. Now ComfyUI fails to import the whole suite unless you kill the program, manually activate its venv, and then `pip install llama-cpp-python`

Which is not obvious for a lot of people.

If you go to the subfolder dedicated to the custom node suite and try to manually `pip install -r requirements.txt`, it won't install the packages in the correct ComfyUI venv, so I'm concerned people will get even more confused.

**Comment by xXAdonesXx** [comment]
![grafik](https://github.com/xXAdonesXx/NodeGPT/assets/66518617/d49de108-ace6-4d40-8162-e23d5f090a3d)
The timeout feature is in the TextGeneration. I have added an Update.bat to facilitate node updates. Furthermore, I have simplified the installation of the nodes. Now, you only need to git clone, and the requirements will install automatically upon the first start of ComfyUI. I have also added an auto-update feature, but it slows the startup, so I have set the default to false.
